pkg/defaults/defaults.go

```diff
@@ -163,6 +168,12 @@ func FromConfig(
 
 		} else if rawStep.TestStepConfiguration != nil {
 			step = steps.TestStep(*rawStep.TestStepConfiguration, config.Resources, podClient, artifactDir, jobSpec)
+		} else {
+			return nil, nil, fmt.Errorf("unhandled rawStep %s", rawStep)
+		}
+
+		if step == nil {
```

Comment:
why?

Reply:
Because my step is only added if there's a pull-request currently.

Reply:
a `nil` step is not supposed to happen

Reply:
It happens in the case I've added since the step isn't added when the job is a non-PR job.

Reply:
I could move the `continue` into my branch  at line 146 I guess, but I still think we should check `if step == nil` and panic if it is in that case, as that would be a programmer error.